.github/workflows/docker-image.yml

```diff
@@ -1,23 +1,38 @@
 name: Docker Image
 
 on:
-  push:
-    branches:
-      - master
+  release:
+    types: [created]
 
 jobs:
   publish:
     name: Build and publish
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          - 36
+          - 37
+          - 38
+    env:
+      IMAGE: openmined/tenseal
 
     steps:
       - uses: actions/checkout@v2
-      
-      - uses: docker/build-push-action@v1
-        with:
-          username: ${{ secrets.DOCKER_LOGIN }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: openmined/tenseal
-          tags: latest
-          tag_with_sha: true
-          add_git_labels: true
+
+      - name: build image
+        run: docker build -t $IMAGE -f docker-images/Dockerfile-py${{ matrix.python-version }} .
+
+      - name: docker login
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" |  docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+
+      - name: tag and push image
+        run: |
+          docker tag $IMAGE $IMAGE:${{ github.event.release.tag_name }}-py${{ matrix.python-version }}
+          docker push $IMAGE:${{ github.event.release.tag_name }}-py${{ matrix.python-version }} .
```

Comment:
Tagging here will depends on whether it's a release or not

Reply:
Fixed.